transportable-udfs-codegen/src/main/java/com/linkedin/transport/codegen/ProjectContext.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.codegen;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.transport.compile.UDFProperties;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+
+/**
+ * Encapsulates configuration related to a Transport UDF project required by a {@link WrapperGenerator} such as
+ * UDF metadata and the output directories for the wrappers and corresponding resources
+ */
+public class ProjectContext {
+
+  private UDFProperties _udfProperties;
+  private File _sourcesOutputDir;
+  private File _resourcesOutputDir;
+
+  public UDFProperties getUdfProperties() {
```

Comment:
Now that UDFProperties is inside ProjectContext, we may need think of a more expressive name since it is too generic and ProjectContext is generic as well. Feel free to ignore this comment if you think naming is good as is.